### PR TITLE
Documentation Update

### PR DESCRIPTION
**Pull request by chamander**
## Abstract:

- Correct sample code regarding `public func equal<T: Equatable>(expectedValue: T?)`. d088ce6
- Introductory work in preferring a more streamline, less bogged down README.
- __Does not__ introduce any __new__ features, and is solely an update to the documentation.

## More Info.:

When attempting to read through the new Predicate API (awesome stuff! thanks @jeffh !), I happened across a documentation mishap regarding the return type of `public func equal(...)`.

It was still `MatcherFunc<T?>`, not yet being updated to `Predicate<T>`.

One thing led to the next and I thought I would go through a good portion of the README while I was at it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -589,25 +610,33 @@ Values must be `Equatable`, `Comparable`, or subclasses of `NSObject`.
...
-Its important to remember that `beIdenticalTo` only makes sense when comparing types with reference semantics, which have a notion of identity. In Swift, that means a `class`. This matcher will not work with types with value semantics such as `struct` or `enum`. If you need to compare two value types, you can either compare individual properties or if it makes sense to do so, make your type implement `Equatable` and use Nimble's equivalence matchers instead.
+It is important to remember that `beIdenticalTo` only makes sense when comparing
+types with reference semantics, which have a notion of identity. In Swift, 
+that means types that are defined as a `class`. 
+
+This matcher will not work when comparing types with value semantics such as
+those defined as a `struct` or `enum`. If you need to compare two value types,
+consider what it means for instances of your type to be identical. This may mean
+comparing individual properties or, if it makes sense to do so, conforming your type 
+to `Equatable` and use Nimble's equivalence matchers instead.
```
Comment (g-ackt) [inline README.md]: Grammar mistake. "using", not "use".

File: README.md [hunk 2]
```diff
@@ -822,40 +853,50 @@ Notes:
...
+expect { try somethingThatThrows() }.to(throwError(NSCocoaError.PropertyListReadCorruptError))
 
-// Passes if somethingThatThrows() throws an error with a given type:
-expect{ try somethingThatThrows() }.to(throwError(errorType: NimbleError.self))
+// Passes if 'somethingThatThrows()' throws an error of a particular type:
+expect { try somethingThatThrows() }.to(throwError(errorType: NimbleError.self))
 ```
 
-If you are working directly with `ErrorProtocol` values, as is sometimes the case when using `Result` or `Promise` types, you can use the `matchError` matcher to check if the error is the same error is is supposed to be, without requiring explicit casting.
+If you are working directly with `Error` values, without explicit casting,
```
Comment (g-ackt) [inline README.md]: Looking to make this much clearer, potentially removing the list the immediately follows.

File: README.md [hunk 3]
```diff
@@ -1025,93 +1074,102 @@ expect(turtles).to(containElementSatisfying(^BOOL(id object) {
...
 ```
 
-For Swift the actual value has to be a Sequence, e.g. an array, a set or a custom seqence type.
+For Swift, the actual value must be an instance of a type conforming to `Sequence`. 
+For example, an array, a set, or a custom sequence type.
 
-For Objective-C the actual value has to be a NSFastEnumeration, e.g. NSArray and NSSet, of NSObjects and only the variant which
+For Objective-C, the actual value must be an instance of a type which implements 
+the `NSFastEnumeration` protocol.
+For example, `NSArray` and `NSSet`, of `NSObject`s and only the variant which
```
Comment (g-ackt) [inline README.md]: I am unsure about what the original author intended this sentence to mean. Particularly the portion of it which references "NSObject".

Comment (g-ackt) [inline README.md]: I am unsure what the original author intended to note here, in this sentence; Particularly the portion of the sentence which references "NSObject".

File: README.md (hunk 1 above)
Comment (chamander) [inline README.md]: Addressed in 0b97b1a

File: README.md (hunk 2 above)
Comment (chamander) [inline README.md]: Addressed in 775093d

File: README.md (hunk 3 above)
Comment (chamander) [inline README.md]: Okay. I think I may have worked it out. This is addressed in 8da0794

I wasn't actually aware that types conforming to NSFastEnumeration were able to capture elements which weren't subtypes of NSObject!

**Comment by jeffh** [comment]
Thanks @chamander for fixing up the README. It's been much needed 😄

**Pull request merged**